File: sintactico/Pila.py

class Pila:
    def __init__(self):
        self.lista = []

    def push(self, x):
        self.lista.insert(0, x)

    def pop(self):
        x = self.lista[0]
        self.lista.pop(0)
        return x

    def top(self):
        return self.lista[0]

    def muestra(self):
        print("Pila: ", end="")
        for x in reversed(self.lista):
            x.muestra()
        print()


if __name__ == "__main__":
    
    pass






class ElementoPila:
    def __init__(self):
        self.simbolo = ""

    def muestra(self):
        pass

    def regresar(self):
        return "0"


class Terminal(ElementoPila):
    def __init__(self, simbolo):
        super().__init__()
        self.simbolo = simbolo

    def muestra(self):
        print(self.simbolo, end="")


class Noterminal(ElementoPila):
    def __init__(self, simbolo):
        super().__init__()
        self.simbolo = simbolo

    def muestra(self):
        print(self.simbolo, end="")


class Estado(ElementoPila):
    def __init__(self, simbolo):
        super().__init__()
        self.simbolo = simbolo

    def muestra(self):
        print(self.simbolo, end="")


class Pila:
    def __init__(self):
        self.lista = []

    def push(self, x):
        self.lista.insert(0, x)

    def top(self):
        return self.lista[0]

    def pop(self):
        x = self.lista[0]
        self.lista.pop(0)
        return x

    def muestra(self):
        print("Pila: ", end="")
        for x in reversed(self.lista):
            x.muestra()
        print()


if __name__ == "__main__":
    
    pass

File: sintactico/Pila.java
package sintactico;

import java.util.LinkedList;
import java.util.List;
import java.util.ListIterator;

public class Pila {
    private List<ElementoPila> lista = new LinkedList<>();

    // Método para agregar un elemento a la pila
    public void push(ElementoPila x) {
        lista.add(0, x);
    }

    // Método para quitar y devolver el elemento superior de la pila
    public ElementoPila pop() {
        ElementoPila x = lista.get(0);
        lista.remove(0);
        return x;
    }

    // Método para obtener el elemento superior de la pila sin quitarlo
    public ElementoPila top() {
        return lista.get(0);
    }

    // Método para mostrar la pila
    public void muestra() {
        ListIterator<ElementoPila> it = lista.listIterator(lista.size());

        ElementoPila x;
        System.out.print("Pila: ");

        while (it.hasPrevious()) {
            x = it.previous();
            x.muestra();
        }

        System.out.println();
    }
}
